gapis/api/vulkan/overdraw.go

```diff
@@ -2197,3 +2200,940 @@ func (*stencilOverdraw) Flush(ctx context.Context, out transform.Writer) error {
+		NewVkImageSubresourceRange(a, // subresourceRange
+			barrierAspectMask, // aspectMask
+			0,                 // baseMipLevel
+			1,                 // levelCount
+			0,                 // baseArrayLayer
+			1,                 // layerCount
+		),
+	)
+	stagingImageToSrcBarrierData := MustAllocData(ctx, s, stagingImageToSrcBarrier)
+
+	// Barrier data for layout transitions of resolve image. This only used when the attachment image is
+	// multi-sampled.
+	resolveImageToDstBarrier := NewVkImageMemoryBarrier(a,
+		VkStructureType_VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER, // sType
+		0, // pNext
+		VkAccessFlags(VkAccessFlagBits_VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT|
+			VkAccessFlagBits_VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT|
+			VkAccessFlagBits_VK_ACCESS_SHADER_WRITE_BIT|
+			VkAccessFlagBits_VK_ACCESS_TRANSFER_WRITE_BIT), // srcAccessMask
+		VkAccessFlags(VkAccessFlagBits_VK_ACCESS_TRANSFER_WRITE_BIT), // dstAccessMask
+		VkImageLayout_VK_IMAGE_LAYOUT_UNDEFINED,                      // oldLayout
+		VkImageLayout_VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,           // newLayout
+		0xFFFFFFFF,     // srcQueueFamilyIndex
+		0xFFFFFFFF,     // dstQueueFamilyIndex
+		resolveImageID, // image
+		NewVkImageSubresourceRange(a, // subresourceRange
+			barrierAspectMask, // aspectMask
+			0,                 // baseMipLevel
+			1,                 // levelCount
+			0,                 // baseArrayLayer
+			1,                 // layerCount
+		),
+	)
+	resolveImageToDstBarrierData := MustAllocData(ctx, s, resolveImageToDstBarrier)
+
+	resolveImageToSrcBarrier := NewVkImageMemoryBarrier(a,
+		VkStructureType_VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER, // sType
+		0, // pNext
+		VkAccessFlags(VkAccessFlagBits_VK_ACCESS_TRANSFER_WRITE_BIT), // srcAccessMask
+		VkAccessFlags(VkAccessFlagBits_VK_ACCESS_TRANSFER_READ_BIT),  // dstAccessMask
+		VkImageLayout_VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,           // oldLayout
+		VkImageLayout_VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,           // newLayout
+		0xFFFFFFFF,     // srcQueueFamilyIndex
+		0xFFFFFFFF,     // dstQueueFamilyIndex
+		resolveImageID, // image
+		NewVkImageSubresourceRange(a, // subresourceRange
+			barrierAspectMask, // aspectMask
+			0,                 // baseMipLevel
+			1,                 // levelCount
+			0,                 // baseArrayLayer
+			1,                 // layerCount
+		),
+	)
+	resolveImageToSrcBarrierData := MustAllocData(ctx, s, resolveImageToSrcBarrier)
+
+	// Barrier data for layout transitions of attachment image
+	attachmentImageToSrcBarrier := NewVkImageMemoryBarrier(a,
+		VkStructureType_VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER, // sType
+		0, // pNext
+		VkAccessFlags( // srcAccessMask
+			VkAccessFlagBits_VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT|
+				VkAccessFlagBits_VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT|
+				VkAccessFlagBits_VK_ACCESS_SHADER_WRITE_BIT|
+				VkAccessFlagBits_VK_ACCESS_TRANSFER_WRITE_BIT|
+				VkAccessFlagBits_VK_ACCESS_TRANSFER_READ_BIT,
```

Comment:
Don't need (and should not specify) read bits in src access mask.

Reply:
This is unchanged from previously (moved from another file). We can fix as a follow-up if thats ok with you.